### PR TITLE
Default dependencies in mjmlconfig

### LINKED ISSUE #2141: TypeError: Cannot convert undefined or null to object
**Description**
When registering a custom component that does not contain a `dependencies` property, the `mjml-validator` gives a very cryptic error,
```
Error when registering custom component :  /path/to/CustomComp.js TypeError: Cannot convert undefined or null to object
```

**To Reproduce**
1. Create a custom mjml component as the default export
```js
export default CustomComp extends BaseComponent {
  // public dependencies = {}; Leave this property out
  public allowedAttributes = {
    attr1: 'string'
  };

  // ...
}
```
2. Add the component to your `.mjmlconfig`
3. Create an mjml template that uses your custom component
4. Compile the mjml with the node SDK
```js
import mjml2html from 'mjml';

const { html, errors } = mjml2html(mjmlFile);

console.log(errors);
```

**Expected behavior**
One of the following...

1. Allow a component to be defined and validated without specifying dependencies.
2. Throw a more descriptive error when registering dependencies that don't exist.

**MJML environment (please complete the following information):**
 - OS: MacOS
 - MJML Version: 4.8.1
 - MJML tool used: Node

**Additional context**
[This line of code](https://github.com/mjmlio/mjml/blob/master/packages/mjml-core/src/helpers/mjmlconfig.js#L95) is where you should handle either 
1. passing an empty object if `.dependencies` is not defined
2. throwing an error telling you that `dependencies` are required.

Without this, the `sources` param of `assignDependencies` ends up being `[null]`, which throws a hard type error when trying to iterate over the items.
https://github.com/mjmlio/mjml/blob/master/packages/mjml-validator/src/dependencies.js#L1

### PR DESCRIPTION
**Pull request by ryanburr**
Fixes #2141 

- Defaults the dependencies of a component to an empty object
- Adds type checks and warnings to dependency assignment

### CODE REVIEW

File: packages/mjml-validator/src/dependencies.js [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
-    for (const tag of Object.keys(source)) {
-      const list = []
-      if (target[tag]) {
-        list.push(...target[tag])
-      }
-      if (source[tag]) {
-        list.push(...source[tag])
+    if (typeof source === 'object' && source !== null) {
+      for (const tag of Object.keys(source)) {
+      if (typeof tag !== 'string') {
```
Comment (TrySound) [inline packages/mjml-validator/src/dependencies.js:9]: Inverse check and fix formatting please

Comment (ryanburr) [reply packages/mjml-validator/src/dependencies.js:9]: Good catch.

**Comment by iRyusa** [comment]
Looks good to me @TrySound are you OK too ?

File: packages/mjml-validator/src/dependencies.js [hunk 2]
```diff
@@ -1,14 +1,26 @@
 export const assignDependencies = (target, ...sources) => {
+  if (!sources) {
```
Comment (TrySound) [inline packages/mjml-validator/src/dependencies.js:2]: Better check length. `sources` is always an array

**Comment by TrySound** [comment]
Last small tweak

File: packages/mjml-validator/src/dependencies.js (hunk 2 above)
Comment (ryanburr) [reply packages/mjml-validator/src/dependencies.js]: Done.

File: packages/mjml-validator/src/dependencies.js [hunk 3]
```diff
@@ -1,14 +1,26 @@
 export const assignDependencies = (target, ...sources) => {
+  if (!sources || !sources.length) {
```
Comment (TrySound) [inline packages/mjml-validator/src/dependencies.js:2]: I mean `if (sources.length === 0)`

Comment (ryanburr) [reply packages/mjml-validator/src/dependencies.js]: Done

**Pull request merged**